Assets/Scripts/RoadControl.cs

```diff
@@ -355,125 +591,467 @@ void Update()
-                        bb.spriteTransform.position = new Vector3(x + (bb.offsetX + roadWidthOffset) * nearEdgeWidthScale,
-                                                                    nearEdgeHeight + bb.spriteType.rect.height * nearEdgeWidthScale / 2, -1f);
-                        bb.spriteRend.enabled = true;
-                    }
+                    Vector2 lineBasePos = warningRoadLines[curWarningRoadLineIndex].rectTransform.anchoredPosition;
+                    warningRoadLines[curWarningRoadLineIndex].rectTransform.anchoredPosition = 
+                                                    new Vector2(warningRoadLinesInitialX + x * roadLineToWarningRoadLineConversion, lineBasePos.y);
+                    warningRoadLines[curWarningRoadLineIndex].rectTransform.sizeDelta = 
+                                                    new Vector2(warningRoadLineBaseWidth * nearEdgeWidthScale, 1.0f);
+                    curWarningRoadLineIndex++;
                 }
-            }
 
-            if (visibleDroids.Count > 0) {
-                foreach (BombDroid bd in visibleDroids) {
+                if (segNumDrawn < segIndexSkipSegThreshold) {
+                    if (visibleBillboards.Count > 0) {
+                        foreach (BillboardSprite bb in visibleBillboards) {
+                            if (bb.segmentIndex == i) {
+
+                                float roadWidthOffset;
+                                if (bb.offsetX < 0) { roadWidthOffset = screenHalfWidth * -1; }
+                                else { roadWidthOffset = screenHalfWidth; }
 
-                    if (bd.segmentIndex == i) {
+                                float sizeCorrection = nearEdgeWidthScale * billboardSizeCorrectionFactor;
 
-                        float roadWidthOffset;
-                        if (bd.offsetX < 0) { roadWidthOffset = screenHalfWidth * -1; }
-                        else { roadWidthOffset = screenHalfWidth; }
+                                float spriteZ;
+                                float spriteElevation = nearEdgeHeight + bb.spriteType.rect.height * nearEdgeWidthScale / 2;
+                                if (nearEdgeHeight < absHighestScreenLineDrawn && roadSegIndexOfHighestScreenLine < i) { spriteZ = spriteBgdZ; }
+                                else { spriteZ = -1f; }
 
-                        float sizeCorrection = nearEdgeWidthScale * billboardSizeCorrectionFactor;
+                                bb.spriteTransform.localScale = new Vector3(sizeCorrection, sizeCorrection, 1);
+                                bb.spriteTransform.position = new Vector3(x + (bb.offsetX) * nearEdgeWidthScale,
+                                                                            spriteElevation, spriteZ);
+                                bb.spriteRend.enabled = true;
+                            }
+                        }
+                    }
```

Comment:
-Remove redundant memory allocations by pre-allocating and reusing variables initialized before the `for` loop instead of calling `new` inside of the loops. As much as I wish we could set the position and localScale directly (for even better efficiency), we cannot assign private variables and have to assign the entire vector instead. But we can avoid redundant memory allocations inside of our loops.

-Don't multiply to take the negative; implicitly add (subtract) via assignment instead (ex. `x = -y` instead of `x = y * -1f`)

```
if (segNumDrawn < segIndexSkipSegThreshold) {
    Vector3 localScale = Vector3.one;
    Vector3 position = Vector3.zero;
    float roadWidthOffset;
    float sizeCorrection = nearEdgeWidthScale * billboardSizeCorrectionFactor;
    float spriteZ;
    float spriteElevation;
    
    if (visibleBillboards.Count > 0) {
        foreach (BillboardSprite bb in visibleBillboards) {
            if (bb.segmentIndex == i) {
                if (bb.offsetX < 0) { roadWidthOffset = -screenHalfWidth; }
                else { roadWidthOffset = screenHalfWidth; }
    
                spriteElevation = nearEdgeHeight + bb.spriteType.rect.height * nearEdgeWidthScale * 0.5f;
    
                if (nearEdgeHeight < absHighestScreenLineDrawn && roadSegIndexOfHighestScreenLine < i) { spriteZ = spriteBgdZ; }
                else { spriteZ = -1f; }
    
                localScale.Set(sizeCorrection, sizeCorrection, 1f);
                bb.spriteTransform.localScale = localScale;
    
                position.Set(x + (bb.offsetX) * nearEdgeWidthScale, spriteElevation, spriteZ);
                bb.spriteTransform.position = position;
    
                bb.spriteRend.enabled = true;
            }
        }
    }
    .
    .
    .
```